overlord/snapstate/autorefresh.go

```diff
@@ -797,6 +810,37 @@ func inhibitRefresh(st *state.State, snapst *SnapState, snapsup *SnapSetup, info
 	return busyErr
 }
 
+func processInhibitedAutoRefresh(chg *state.Change, old state.Status, new state.Status) {
+	if !new.Ready() || chg.Kind() != "auto-refresh" {
+		return
+	}
+
+	if err := maybeAddRefreshInhibitNotice(chg.State()); err != nil {
+		logger.Debugf(`internal error: failed to add "refresh-inhibit" notice: %v`, err)
+		return
+	}
+}
+
+func maybeAddRefreshInhibitNotice(st *state.State) error {
+	snapStates, err := All(st)
+	if err != nil {
+		return err
+	}
+
+	for _, snapst := range snapStates {
+		if snapst.RefreshInhibitedTime == nil {
+			continue
+		}
+		// One inhibited snap is enough to send the notice.
+		if _, err := st.AddNotice(nil, state.RefreshInhibitNotice, "-", nil); err != nil {
```

Comment:
Is there any notice issued when the snap becomes uninhibited? If the key is always `"-"`, it seems like there would be no way to know. If there were a key for "inhibited" and a key for "uninhibited", then whichever one was more recent would be the current inhibition state.

Alternatively, this could be encoded in the notice data.

But perhaps we don't want to have this kind of information. If that is the case, why not?

Reply:
Ahh, probably because of race conditions. Need to check inhibited status and get lock directly, in case it becomes {,un}inhibited between when the notice is received and when that information is used. So the notice then is just saying "inhibited status changed, go query it". Am I understanding correctly?

Reply:
In the spec we don't need to track snaps getting uninhibited, only in very special cases when a refresh is forced, and in that case a change-update notice + api query can be used as mentioned in the spec.

The intention behind this notice is exactly like you mentioned "inhibited status changed, go query it" specifically querying [refresh-inhibit-proceed-time](https://github.com/snapcore/snapd/pull/13580) `/v2/snaps` . "-" is used intentionally (because it is not a valid snap-name) to indicate that this notice is not about a specific snap, allowing us to later use this key to indicate inhibition status  changes per-snap if we ever need to do that.